src/OrchardCore.Modules/OrchardCore.Autoroute/Routing/AutorouteRoute.cs

```diff
@@ -44,31 +49,57 @@ public VirtualPathData GetVirtualPath(VirtualPathContext context)
                         }
                     }
 
-                    return new VirtualPathData(_target, path );
+                    return new VirtualPathData(_target, path);
                 }
             }
+            
 
             return null;
         }
 
-        public Task RouteAsync(RouteContext context)
+        public async Task RouteAsync(RouteContext context)
         {
             var requestPath = context.HttpContext.Request.Path.Value;
 
-            string contentItemId;
-
-            if(_entries.TryGetContentItemId(requestPath, out contentItemId))
+            if (_entries.TryGetContentItemId(requestPath, out var contentItemId))
             {
-                context.RouteData.Values["area"] = "OrchardCore.Contents";
-                context.RouteData.Values["controller"] = "Item";
-                context.RouteData.Values["action"] = "Display";
-                context.RouteData.Values["contentItemId"] = contentItemId;
+                await EnsureRouteData(context, contentItemId);
+                await _target.RouteAsync(context);
+            }
+        }
 
-                context.RouteData.Routers.Add(_target);
-                return _target.RouteAsync(context);
+        private async Task<RouteValueDictionary> GetContentItemDisplayRoutes(HttpContext context, string contentItemId)
+        {
+            if (string.IsNullOrEmpty(contentItemId))
+                return null;
+            var contentManager = context.RequestServices.GetService<IContentManager>();
+            var contentItem = await contentManager.GetAsync(contentItemId);
+            return contentItem == null ? null : contentManager.PopulateAspect<ContentItemMetadata>(contentItem)?.DisplayRouteValues;
+        }
+
+        private async Task EnsureRouteData(RouteContext context, string contentItemId)
+        {
+            var displayRoutes = await GetContentItemDisplayRoutes(context.HttpContext, contentItemId);
+            if (displayRoutes == null)
```

Comment:
If the default routes are in a default handler, then just don't match anything when it's null.

Reply:
fixed